pkgs/development/libraries/glibc/ld-fallback.patch

```diff
@@ -0,0 +1,465 @@
+From 9d7cc3850df04c15d9c82e51c9e9198ead7bf847 Mon Sep 17 00:00:00 2001
+From: Jan Malakhovski <oxij@oxij.org>
+Date: Mon, 14 Aug 2023 13:00:00 +0000
+Subject: [PATCH] Implement support for LD_FALLBACK_PATH envvar and related
+ command line options
+
+LD_FALLBACK_PATH has the same semantics as LD_LIBRARY_PATH but unlike
+LD_LIBRARY_PATH, instead of having the highest priority, it has the
+second-lowest (it gets used just before the default system directories).
+
+Thus, LD_FALLBACK_PATH provides a way to specify fallback library paths.
+
+In other words, LD_FALLBACK_PATH provides a way to override system libraries
+without creating a chroot.
+
+Consider the following use cases:
+
+* Sometimes you just want to change a set of system libraries for a set of
+  programs without changing the system default. For instance, if you have
+  different GPUs requiring different libGL, CUDA, and/or OpenCL
+  implementations. (Though, specifically for libGL there are better solutions
+  for per-screen routing of OpenGL calls.)
```

Comment:
> And the other issue ("change a set of system libraries for a set of
programs without changing the system default") doesn't seem applicable in the context of NixOS where DT_RUNPATH is used in build products which takes precedence over LD_FALLBACK_PATH. Or am I missing something here? @Ma27 

I think this is just a slightly misleading wording. The comment specifically enumerates various impurely-linked drivers, so I presume that is what was meant by "system libraries"